### DISCUSSION TITLE
GenericEnumMapper is a bit too chatty on the logs for me

### DISCUSSION THREAD

**Discussion by razzmatazz** [nhibernate/fluent-nhibernate#563] [Q&A] [answered]
Hi and thanks for the awesome lib!

One issue I have is GenericEnumMapper<> (or related code) is that it is emitting quite a lot of debug messages when it performs conversions string -> enum (and presumably back). This makes reading system logs a bit cumbersome.

For example (with assembly, enum and table names replaced to protect the innocent):
```
DEBUG FluentNHibernate.Mapping.GenericEnumMapper`1[[(assembly)]], returning null as column: table10_68_0_
DEBUG FluentNHibernate.Mapping.GenericEnumMapper`1[[(assembly)]], binding 'EnumValue' to parameter: 8
DEBUG FluentNHibernate.Mapping.GenericEnumMapper`1[[(assembly)]], returning 'EnumValue' as column: table10_68_1_
```

The mapping is
```csharp
Map(x => x.EnumProp, "enum_column").Nullable().Length(64).CustomType<GenericEnumMapper<MyEnum>>();
```

Maybe the mapping is not how it should look for mapping an enum value to string on DB?

However I could not find the code that does this in the repo that does this logging with simple greps. Maybe it is in some dependency or NHibernate itself? If you could point me to the place I could make a PR then.

Is there a way to tune down the volume of the log messages? I guess one way would be to configure logging system to match on GenericEnumMapper, but maybe it would be good for the default behaviour to be a little bit more quiet?

**Comment by hazzik** [comment, marked as answer]
> However I could not find the code that does this in the repo that does this logging with simple greps.
>  Maybe it is in some dependency or NHibernate itself?

Yes, it is in NHibernate. The code is here:

* `"returning null as column"`: https://github.com/nhibernate/nhibernate-core/blob/47992e73ed6114093f31ed99829dd8c4c7663829/src/NHibernate/Type/NullableType.cs#L222-L225
* `"binding 'EnumValue' to parameter"`: https://github.com/nhibernate/nhibernate-core/blob/47992e73ed6114093f31ed99829dd8c4c7663829/src/NHibernate/Type/NullableType.cs#L157-L160
* `"returning 'EnumValue' as column"` https://github.com/nhibernate/nhibernate-core/blob/47992e73ed6114093f31ed99829dd8c4c7663829/src/NHibernate/Type/NullableType.cs#L245-L248

> Is there a way to tune down the volume of the log messages?

Yes, you should be able to disable debug logging for this logger. Use `"NHibernate.Type"` logger to do so. (The exact code depends on a logging provider you use).

PS: you should not use debug logging in production.
